tracing-opentelemetry/src/fmt.rs

```diff
@@ -0,0 +1,443 @@
+                    }
+
+                    let parent = builder.parent_context.as_ref()?;
+                    let parent_span = parent.span().span_context();
+                    if parent_span.is_remote() {
+                        parent.remote_span_context().map(|cx| cx.trace_id())
+                    } else {
+                        Some(parent_span.trace_id())
+                    }
+                })
+                .unwrap_or_else(otel::TraceId::invalid)
+        };
+
+        let bold = if self.ansi {
+            Style::new().bold()
+        } else {
+            Style::new()
+        };
+
+        let dimmed = if self.ansi {
+            Style::new().dimmed()
+        } else {
+            Style::new()
+        };
+
+        write!(
+            writer,
+            "{}{:032x}{} ",
+            bold.paint("["),
+            trace_id.to_u128(),
+            bold.paint("]")
+        )?;
+
+        self.format_timestamp(writer)?;
+
+        self.format_level(*meta.level(), writer)?;
+
+        if self.display_thread_name {
+            let current_thread = std::thread::current();
+            match current_thread.name() {
+                Some(name) => {
+                    write!(writer, "{} ", FmtThreadName::new(name))?;
+                }
+                // fall-back to thread id when name is absent and ids are not enabled
+                None if !self.display_thread_id => {
+                    write!(writer, "{:0>2?} ", current_thread.id())?;
+                }
+                _ => {}
+            }
+        }
+
+        if self.display_thread_id {
+            write!(writer, "{:0>2?} ", std::thread::current().id())?;
+        }
+
+        let mut wrote_spans = false;
+        for span in ctx.scope() {
+            let meta = span.metadata();
+            let ext = span.extensions();
+            let fields = &ext
+                .get::<FormattedFields<N>>()
+                .expect("Unable to find FormattedFields in extensions; this is a bug");
+
+            if let Some(builder) = ext.get::<otel::SpanBuilder>() {
+                if self.display_remote_parents {
+                    if let Some(remote_parent) = builder
+                        .parent_context
+                        .as_ref()
+                        .and_then(TraceContextExt::remote_span_context)
+                    {
+                        write!(writer, "{}", dimmed.prefix())?;
+                        if self.display_span_ids {
+                            write!(writer, "[{:x}]", remote_parent.span_id().to_u64())?;
+                        }
+                        // TODO(eliza): it would be nice if we could not print
+                        // anything here if the remote parent's trace state is
+                        // empty...but there's no way to check currently.
+                        write!(
+                            writer,
+                            "{{{:?}}}{}:",
+                            remote_parent.trace_state(),
```

Comment:
`remote_parent.trace_flags()` may actually be more interesting here as it contains sampling information and is present in all spans, values from the [tracestate-header](https://www.w3.org/TR/trace-context/#tracestate-header) are less common and often opaque to the application anyway

Reply:
okay, thanks! i wasn't sure which data was more likely to be useful — perhaps we would want an option to show trace flags for local spans as well, for debugging sampling?

i was _hoping_ there was a way to get key-value data, names, etc for remote spans to display in the logs, but it seems like that's not available to us...(which does make sense; we wouldn't want to pass that data around everywhere). is there any way we can hydrate that data from span IDs, or is that not supported?

Reply:
> okay, thanks! i wasn't sure which data was more likely to be useful — perhaps we would want an option to show trace flags for local spans as well, for debugging sampling?

Yeah I think that would make sense as an option 👍 

> i was hoping there was a way to get key-value data, names, etc for remote spans to display in the logs, but it seems like that's not available to us...(which does make sense; we wouldn't want to pass that data around everywhere). is there any way we can hydrate that data from span IDs, or is that not supported?

That would be interesting but I don't think that it's possible to get information about remote spans currently, you basically just work with a serialized `SpanContext` when injecting/extracting.